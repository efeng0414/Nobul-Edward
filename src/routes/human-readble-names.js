export const humanReadableNameKeys = {
  browseListingsMap: "home.browseProperties",
  myFavorites: "myFavorites",
  taggedListings: "consumerViewTaggedListings",
  back: "goBack"
};
