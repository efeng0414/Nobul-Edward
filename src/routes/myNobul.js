import loadable from "loadable-components";

/**************************
    Consumer components
/**************************/

export const ConsumerSettings = loadable(() =>
  import("../scenes/consumer-settings")
);

export const Notifications = loadable(() => import("../scenes/notifications"));

export const ConsumerJobs = loadable(() => import("../scenes/consumer-jobs"));

export const Contracts = loadable(() => import("../scenes/contracts"));

export const AgentOffers = loadable(() => import("../scenes/agent-offers"));

export const Subscriptions = loadable(() =>
  import("../scenes/agent-subscriptions")
);

export const ConsumerOffers = loadable(() =>
  import("../scenes/consumer-offers")
);

export const ConsumerJobDetails = loadable(() =>
  import("../scenes/consumer-job-details")
);

export const ConsumerJobOffers = loadable(() =>
  import("../scenes/consumer-job-offers")
);

export const ConsumerMyFavorites = loadable(() =>
  import("../scenes/consumer-my-favorites")
);

export const ConsumerViewTaggedListings = loadable(() =>
  import("../scenes/consumer-view-tagged-listings")
);

export const GenerateAgreement = loadable(() =>
  import("../scenes/create-agreement")
);

export const ConsumerViewMyEvents = loadable(() =>
  import("../scenes/consumer-view-my-events")
);

export const ConsumerOfferDetails = loadable(() =>
  import("../scenes/consumer-offer-details")
);

export const ConsumerProposalDetails = loadable(() =>
  import("../scenes/consumer-proposal-details")
);

export const NotFound = loadable(() => import("../scenes/not-found"));

/**************************
    Agent components
/**************************/

export const AgentSettings = loadable(() => import("../scenes/agent-settings"));

export const AgentViewMyEvents = loadable(() =>
  import("../scenes/agent-view-my-events")
);

export const AgentOfferDetails = loadable(() =>
  import("../scenes/agent-offer-details")
);

/**************************
    Unused components
/**************************/

// Used in contracts to switch paths.
export const contractPaths = {
  generate: "generate",
  actionRequired: "action-required",
  waitingForOthers: "waiting",
  completed: "completed"
};

/**************************
    Routes
/**************************/
export const url = {
  root: "/my-dashboard",
  settings: "/my-dashboard/settings",
  contracts: "/my-dashboard/contracts",
  jobs: "/my-dashboard/postings",
  offers: "/my-dashboard/proposals",
  myFavorites: "/my-dashboard/favourites",
  events: "/my-dashboard/events",
  subscriptions: "/my-dashboard/subscriptions",
  notifications: "/my-dashboard/activities",
  consumerViewTaggedListings: "/my-dashboard/tagged-properties",
  consumerViewMyEvents: "/my-dashboard/events",
  offerDetails: "/my-dashboard/proposals/:jobType/:offerId",
  taggedListings: "/my-dashboard/tagged-properties",

  // Contract
  contractDefault: "/my-dashboard/contracts/:path?",
  contractGenerate: `/my-dashboard/contracts/${contractPaths.generate}`,
  contractActionRequired: `/my-dashboard/contracts/${
    contractPaths.actionRequired
  }`,
  contractWaitingForOthers: `/my-dashboard/contracts/${
    contractPaths.waitingForOthers
  }`,
  contractCompleted: `/my-dashboard/contracts/${contractPaths.completed}`,

  //not currently linked in my-dashboard page
  consumerJobDetails: "/my-dashboard/postings/:jobType/:jobId",
  consumerJobOffers: "/my-dashboard/postings/:jobType/:jobId/proposals",
  generateAgreementRoot: "/my-dashboard/generate-agreement",
  generateAgreement:
    "/my-dashboard/generate-agreement/:format?/:formType?/:offerId?",

  //partial url
  partialConsumerJobDetails: "/my-dashboard/postings",
  partialConsumerJobOffers: "/my-dashboard/proposals",
  accountSettings: "/my-dashboard/settings?tab=account"
};

/**************************
    Routes helper functions
/**************************/
export const getUserPostingUrl = ({ jobType, jobId }) => {
  return url.consumerJobDetails
    .replace(":jobType", jobType)
    .replace(":jobId", jobId);
};

export const getAgentProposalUrl = ({ jobType, offerId }) => {
  return url.offerDetails
    .replace(":jobType", jobType)
    .replace(":offerId", offerId);
};
