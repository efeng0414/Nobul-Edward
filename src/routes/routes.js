import loadable from "loadable-components";

export const NotFound = loadable(() => import("../scenes/not-found"));

export const PermissionDenied = loadable(() =>
  import("../scenes/permission-denied")
);
export const HomePage = loadable(() => import("../scenes/home-page"));

export const BlockEU = loadable(() => import("../scenes/block-eu"));

export const ConsumerRegistration = loadable(() =>
  import("../scenes/consumer-registration")
);

export const AgentRegistration = loadable(() =>
  import("../scenes/agent-registration")
);

export const StandardProposalFlow = loadable(() =>
  import("../scenes/standard-proposal-flow")
);

export const CreateBuyJob = loadable(() => import("../scenes/create-buy-job"));
export const CreateSellJob = loadable(() =>
  import("../scenes/create-sell-job")
);
export const NobulPremium = loadable(() => import("../scenes/nobul-premium"));
export const AgentSettings = loadable(() => import("../scenes/agent-settings"));
export const ConsumerSettings = loadable(() =>
  import("../scenes/consumer-settings")
);

export const MarketPlace = loadable(() => import("../scenes/market-place"));
export const ShowDraftJob = loadable(() => import("../scenes/show-draft-job"));

export const AgentUpdateOffersDetails = loadable(() =>
  import("../scenes/agent-update-offers-details")
);

export const EditJob = loadable(() => import("../scenes/edit-job"));
export const BrowseListingsMap = loadable(() =>
  import("../scenes/browse-listings-map")
);
export const BrowseListings = loadable(() =>
  import("../scenes/browse-listings")
);

export const ViewAgreement = loadable(() => import("../scenes/view-agreement"));
export const ListingDetails = loadable(() =>
  import("../scenes/listing-details")
);

export const ConsumerMyNobul = loadable(() =>
  import("../scenes/my-nobul-consumer")
);

export const AgentJobDetails = loadable(() =>
  import("../scenes/agent-job-details")
);

export const AgentMyNobul = loadable(() => import("../scenes/my-nobul-agent"));

export const AgentRatingView = loadable(() =>
  import("../scenes/agent-rating-view")
);

export const Notifications = loadable(() => import("../scenes/notifications"));

export const GetStarted = loadable(() => import("../scenes/get-started"));

export const ConsumerHomePage = loadable(() =>
  import("../scenes/consumer-home")
);

export const AgentHomePage = loadable(() => import("../scenes/agent-home"));

export const Landing1 = loadable(() => import("../scenes/landing-1"));
export const Landing2 = loadable(() => import("../scenes/landing-2"));
export const Landing3 = loadable(() => import("../scenes/landing-3"));
export const Landing4 = loadable(() => import("../scenes/landing-4"));
export const Landing5 = loadable(() => import("../scenes/landing-5"));
export const FirebaseAuthActionHandler = loadable(() =>
  import("../scenes/firebase-auth-action-handler")
);

export const ErrorPage = loadable(() => import("../scenes/error-page"));
export const Redirect = loadable(() => import("../components/redirect"));
export const InternalVerification = loadable(() =>
  import("../components/internal-verification")
);

export const partialUrl = {
  landing: "real-estate-for-everyone",
  homePage: "/"
};

export const url = {
  home: "/",
  blockEU: "/nobul-eu",
  // homePage: "/",
  landing1: "/real-estate-for-everyone-01",
  landing2: "/real-estate-for-everyone-02",
  landing3: "/real-estate-for-everyone-03",
  landing4: "/real-estate-for-everyone-04",
  landing5: "/real-estate-for-everyone-05",
  homeWithLogin: "/login",
  howItWorks: "/how-it-works",
  permissionDenied: "/permission-denied",
  createBuyJob: "/buy-real-estate",
  contracts: "/contracts",
  createSellJob: "/sell-real-estate",
  consumerRegistration: "/consumer-create-profile",
  agentRegistration: "/agent-create-profile",
  receiveMeetingInvitation: "/receiveMeetingInvitation",
  nobulPremium: "/nobulPremium",
  agentSettings: "/agentSettings",
  consumerSettings: "/consumerSettings",
  marketPlace: "/marketplace",
  showDraftJob: "/showDraftJob",
  browseListings: "/search-real-estate",
  browseListingsMap: "/search-real-estate-map",
  listingDetails: "/property-details/:listingId",
  agentViewOffers: "/my-dashboard/offers",
  agentViewJobDetails: "/proposals/:jobType/:jobId",
  editJob: "/editJob",
  agentViewOfferDetails: "/agentViewOfferDetails/:jobType/:offerId",
  agentUpdateOfferDetails: "/agentUpdateOfferDetails",
  agentRatingView: "/agent-rating/:jobType/:offerId",
  viewAgreement: "/viewAgreement/:agreementType/:agreementId",
  mortgageCalculator: "/mortgageCalculator",
  getStarted: "/get-started",
  standardProposalFlow: "/standard-proposal-flow/step/:step?/:stage?",
  authActionHandler: "/auth-action",
  firebaseAuthActionHandler: "/__/auth/action",

  // Add myDashboard sub-routes to myDashboard routes file
  myNobul: "/my-dashboard",
  notifications: "/myDashboard/activities",

  //agent my nobul
  agentMyNobul: "/agentMyNobul",
  error: "/error",
  internalVerification: "/internal-verification",
  redirect: "/r/:page"
};

export const externalLinks = {
  learnRealEstate: "https://wp.nobul.com/"
};
