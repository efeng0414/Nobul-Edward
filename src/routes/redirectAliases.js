import * as myNobul from "./myNobul";
import { url } from "./routes";
import {
  EMAIL_WELCOME_BROWSE_PROPERTIES_TOP,
  EMAIL_WELCOME_FIND_AGENT_TOP,
  EMAIL_WELCOME_BROWSE_PROPERTIES_BOTTOM,
  EMAIL_WELCOME_FIND_AGENT_BOTTOM,
  EMAIL_WELCOME_FACEBOOK,
  EMAIL_WELCOME_TWITTER,
  EMAIL_WELCOME_INSTAGRAM,
  EMAIL_WELCOME_LINKEDIN,
  EMAIL_BIWEEKLY_INCOMPLETE_COMPLETEPROFILE_TOP,
  EMAIL_BIWEEKLY_INCOMPLETE_COMPLETEPROFILE_MIDDLETOPLEFT,
  EMAIL_BIWEEKLY_INCOMPLETE_FAVORITE_MIDDLETOPRIGHT,
  EMAIL_BIWEEKLY_INCOMPLETE_POSTING_MIDDLEBOTTOMLEFT,
  EMAIL_BIWEEKLY_INCOMPLETE_TAGPROPERTY_MIDDLEBOTTOMRIGHT,
  EMAIL_BIWEEKLY_INCOMPLETE_BROWSE_BOTTOM,
  EMAIL_BIWEEKLY_INCOMPLETE_FACEBOOK,
  EMAIL_BIWEEKLY_INCOMPLETE_TWITTER,
  EMAIL_BIWEEKLY_INCOMPLETE_INSTAGRAM,
  EMAIL_BIWEEKLY_INCOMPLETE_LINKEDIN,
  EMAIL_BIWEEKLY_BUYER_POSTING_TOP,
  EMAIL_BIWEEKLY_BUYER_FAVORITE_MIDDLETOPLEFT,
  EMAIL_BIWEEKLY_BUYER_POSTING_MIDDLETOPRIGHT,
  EMAIL_BIWEEKLY_BUYER_TAGPROPERTY_MIDDLEBOTTOMLEFT,
  EMAIL_BIWEEKLY_BUYER_PROPOSALS_MIDDLEBOTTOMRIGHT,
  EMAIL_BIWEEKLY_BUYER_POSTING_BOTTOM,
  EMAIL_BIWEEKLY_BUYER_FACEBOOK,
  EMAIL_BIWEEKLY_BUYER_TWITTER,
  EMAIL_BIWEEKLY_BUYER_INSTAGRAM,
  EMAIL_BIWEEKLY_BUYER_LINKEDIN,
  EMAIL_BIWEEKLY_SELLER_POSTING_TOP,
  EMAIL_BIWEEKLY_SELLER_FAVORITE_MIDDLETOPLEFT,
  EMAIL_BIWEEKLY_SELLER_POSTING_MIDDLETOPRIGHT,
  EMAIL_BIWEEKLY_SELLER_TAGPROPERTY_MIDDLEBOTTOMLEFT,
  EMAIL_BIWEEKLY_SELLER_PROPOSALS_MIDDLEBOTTOMRIGHT,
  EMAIL_BIWEEKLY_SELLER_POSTING_BOTTOM,
  EMAIL_BIWEEKLY_SELLER_FACEBOOK,
  EMAIL_BIWEEKLY_SELLER_TWITTER,
  EMAIL_BIWEEKLY_SELLER_INSTAGRAM,
  EMAIL_BIWEEKLY_SELLER_LINKEDIN
} from "../utilities/google-tag-variable";

// Use route /r/[path] to redirect to the required URL
export const redirectAliases = {
  /******************************** 
      Internal links - use url object
    *********************************/

  // bi-weekly: seg 1, incomplete
  "email-biweekly-incomplete-completeprofile-top": {
    path: `${myNobul.url.settings}?tab=account`,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_COMPLETEPROFILE_TOP
  },
  "email-biweekly-incomplete-completeprofile-middletopleft": {
    path: `${myNobul.url.settings}?tab=account`,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_COMPLETEPROFILE_MIDDLETOPLEFT
  },
  "email-biweekly-incomplete-favorite-middletopright": {
    path: myNobul.url.myFavorites,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_FAVORITE_MIDDLETOPRIGHT
  },
  "email-biweekly-incomplete-posting-middlebottomleft": {
    path: `${url.createBuyJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_POSTING_MIDDLEBOTTOMLEFT
  },
  "email-biweekly-incomplete-tagproperty-middlebottomright": {
    path: myNobul.url.taggedListings,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_TAGPROPERTY_MIDDLEBOTTOMRIGHT
  },
  "email-biweekly-incomplete-browse-bottom": {
    path: url.browseListings,
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_BROWSE_BOTTOM
  },

  // bi-weekly: seg 2, buyer
  "email-biweekly-buyer-posting-top": {
    path: `${url.createBuyJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_POSTING_TOP
  },
  "email-biweekly-buyer-favorite-middletopleft": {
    path: myNobul.url.myFavorites,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_FAVORITE_MIDDLETOPLEFT
  },
  "email-biweekly-buyer-posting-middletopright": {
    path: `${url.createBuyJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_POSTING_MIDDLETOPRIGHT
  },
  "email-biweekly-buyer-tagproperty-middlebottomleft": {
    path: myNobul.url.taggedListings,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_TAGPROPERTY_MIDDLEBOTTOMLEFT
  },
  "email-biweekly-buyer-proposals-middlebottomright": {
    path: url.getStarted,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_PROPOSALS_MIDDLEBOTTOMRIGHT
  },
  "email-biweekly-buyer-posting-bottom": {
    path: `${url.createBuyJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_POSTING_BOTTOM
  },

  // bi-weekly: seg 2, seller
  "email-biweekly-seller-posting-top": {
    path: `${url.createSellJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_POSTING_TOP
  },
  "email-biweekly-seller-favorite-middletopleft": {
    path: myNobul.url.myFavorites,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_FAVORITE_MIDDLETOPLEFT
  },
  "email-biweekly-seller-posting-middletopright": {
    path: `${url.createSellJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_POSTING_MIDDLETOPRIGHT
  },
  "email-biweekly-seller-tagproperty-middlebottomleft": {
    path: myNobul.url.taggedListings,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_TAGPROPERTY_MIDDLEBOTTOMLEFT
  },
  "email-biweekly-seller-proposals-middlebottomright": {
    path: url.getStarted,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_PROPOSALS_MIDDLEBOTTOMRIGHT
  },
  "email-biweekly-seller-posting-bottom": {
    path: `${url.createSellJob}?ref=landing`,
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_POSTING_BOTTOM
  },

  // welcome email
  "email-welcome-browse-properties-top": {
    path: url.browseListings,
    GTM_EVENT: EMAIL_WELCOME_BROWSE_PROPERTIES_TOP
  },
  "email-welcome-find-agent-top": {
    path: url.getStarted,
    GTM_EVENT: EMAIL_WELCOME_FIND_AGENT_TOP
  },
  "email-welcome-browse-properties-bottom": {
    path: url.browseListings,
    GTM_EVENT: EMAIL_WELCOME_BROWSE_PROPERTIES_BOTTOM
  },
  "email-welcome-find-agent-bottom": {
    path: url.getStarted,
    GTM_EVENT: EMAIL_WELCOME_FIND_AGENT_BOTTOM
  },
  "email-welcome-dashboard-footer": {
    path: url.myNobul
  },
  "email-welcome-find-agent-footer": {
    path: url.getStarted
  },
  "email-welcome-browse-properties-footer": {
    path: url.browseListings
  },

  /**************************************** 
     External links - must begin with "http"
    *****************************************/

  // bi-weekly: seg 1, incomplete
  "email-biweekly-incomplete-facebook": {
    path: "http://www.facebook.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_FACEBOOK
  },
  "email-biweekly-incomplete-linkedin": {
    path: "https://www.linkedin.com/company/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_LINKEDIN
  },
  "email-biweekly-incomplete-twitter": {
    path: "https://twitter.com/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_TWITTER
  },
  "email-biweekly-incomplete-instagram": {
    path: "http://www.instagram.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_INCOMPLETE_INSTAGRAM
  },
  "email-biweekly-incomplete-contact-bottom": {
    path: "https://corp.nobul.com/contact-us"
  },

  // bi-weekly: seg 2, buyer
  "email-biweekly-buyer-facebook": {
    path: "http://www.facebook.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_FACEBOOK
  },
  "email-biweekly-buyer-linkedin": {
    path: "https://www.linkedin.com/company/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_LINKEDIN
  },
  "email-biweekly-buyer-twitter": {
    path: "https://twitter.com/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_TWITTER
  },
  "email-biweekly-buyer-instagram": {
    path: "http://www.instagram.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_BUYER_INSTAGRAM
  },
  "email-biweekly-buyer-contact-bottom": {
    path: "https://corp.nobul.com/contact-us"
  },

  // bi-weekly: seg 2, seller
  "email-biweekly-seller-facebook": {
    path: "http://www.facebook.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_FACEBOOK
  },
  "email-biweekly-seller-linkedin": {
    path: "https://www.linkedin.com/company/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_LINKEDIN
  },
  "email-biweekly-seller-twitter": {
    path: "https://twitter.com/nobul",
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_TWITTER
  },
  "email-biweekly-seller-instagram": {
    path: "http://www.instagram.com/nobulrealestate",
    GTM_EVENT: EMAIL_BIWEEKLY_SELLER_INSTAGRAM
  },
  "email-biweekly-seller-contact-bottom": {
    path: "https://corp.nobul.com/contact-us"
  },

  // welcome email
  "email-welcome-twitter": {
    path: "https://twitter.com/nobul",
    GTM_EVENT: EMAIL_WELCOME_TWITTER
  },
  "email-welcome-facebook": {
    path: "http://www.facebook.com/nobulrealestate",
    GTM_EVENT: EMAIL_WELCOME_FACEBOOK
  },
  "email-welcome-instagram": {
    path: "http://www.instagram.com/nobulrealestate",
    GTM_EVENT: EMAIL_WELCOME_INSTAGRAM
  },
  "email-welcome-linkedin": {
    path: "https://www.linkedin.com/company/nobul",
    GTM_EVENT: EMAIL_WELCOME_LINKEDIN
  },
  "email-welcome-terms-footer": {
    path: "https://corp.nobul.com/legal/terms-and-conditions/"
  },
  "email-welcome-privacy-footer": {
    path: "https://corp.nobul.com/legal/privacy-policy/"
  }
};
