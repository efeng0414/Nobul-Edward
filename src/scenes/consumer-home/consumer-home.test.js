import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { shape } from "prop-types";
import * as Routes from "../../routes/routes";
import ConsumerHomeWrapped from "./container";

describe("ConsumerHome", () => {
  const { WrappedComponent: ConsumerHome } = ConsumerHomeWrapped;
  const containerProps = {
    intl: global.getIntl(),
    promptLoginOnMount: false,
    promptLogin: () => {},
    history: { push: () => {} },
    currentUserId: "",
    open: () => {}
  };

  const containerJSX = <ConsumerHome {...containerProps} />;

  describe("Container Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{containerJSX}</IntlProvider>
    );
    const containerSnapshot = renderer.create(snapshotJSX).toJSON();
    it("should match stored snapshot", () => {
      expect(containerSnapshot).toMatchSnapshot();
    });
  });

  describe("componentRender", () => {
    const router = {
      history: new BrowserRouter().history,
      route: {
        location: {},
        match: {}
      }
    };
    const contextObject = {
      context: { intl: containerProps.intl, router },
      childContextTypes: { intl: intlShape.isRequired, router: shape({}) }
    };

    const functionsToReset = ["handleBuyOrSell", "handleBrowse", "handleLearn"];
    const componentPrototypeResets = functionsToReset.reduce(
      global.createComponentPrototypeResets({ component: ConsumerHome }),
      {}
    );
    const containerMount = global.Enzyme.shallow(containerJSX, contextObject);

    beforeEach(() => {
      functionsToReset.map(
        global.resetPropertyComponentProperty({
          component: ConsumerHome,
          componentMount: containerMount,
          componentPrototypeResets
        })
      );
    });

    const mockHistory = {
      calledWithParam: "",
      push: function(url) {
        this.calledWithParam = url;
      }
    };

    describe("ComponentDidMount", () => {
      it("should call props.promptLogin on mount", () => {
        const promptLoginSpy = jest.fn();
        containerProps.promptLoginOnMount = true;
        containerProps.promptLogin = promptLoginSpy;
        const wrapper = global.Enzyme.shallow(
          <ConsumerHome {...containerProps} />,
          contextObject
        );

        expect(promptLoginSpy).toHaveBeenCalled();
      });
    });

    describe("handleBuyOrSell", () => {
      it("should call handleBuyOrSell function on button Click", () => {
        const mockEvent = {
          preventDefault: function() {
            this.preventDefaultCalled = true;
          },
          preventDefaultCalled: false
        };
        const handleBuyOrSellMock = jest.fn();
        containerMount.instance().handleBuyOrSell = handleBuyOrSellMock;
        containerMount.setProps({ a: 1 });
        containerMount
          .find("Box")
          .at(0)
          .simulate("click", mockEvent);

        expect(handleBuyOrSellMock).toBeCalled();
      });

      it("should call preventDefault on the event", () => {
        const mockEvent = {
          preventDefault: function() {
            this.preventDefaultCalled = true;
          },
          preventDefaultCalled: false
        };
        containerMount.setProps({ b: 1 });
        containerMount
          .find("Box")
          .at(0)
          .simulate("click", mockEvent);
        expect(mockEvent.preventDefaultCalled).toBe(true);
      });

      it("should push url to getStarted route on button click if there is a registered user", () => {
        const mockEvent = {
          preventDefault: function() {
            this.preventDefaultCalled = true;
          },
          preventDefaultCalled: false
        };
        containerMount.setProps({
          history: mockHistory,
          currentUserId: "consumer1234"
        });
        containerMount
          .find("Box")
          .at(0)
          .simulate("click", mockEvent);
        expect(mockHistory.calledWithParam).toBe(Routes.url.getStarted);
      });
    });

    describe("handleBrowseEvent", () => {
      const mockEvent = {
        preventDefault: function() {
          this.preventDefaultCalled = true;
        },
        preventDefaultCalled: false
      };
      it("should call preventDefault on the event", () => {
        containerMount.setProps({ b: 1 });
        containerMount
          .find("Box")
          .at(1)
          .simulate("click", mockEvent);
        expect(mockEvent.preventDefaultCalled).toBe(true);
      });

      it("should push url to BrowseListings on button click", () => {
        containerMount.setProps({
          history: mockHistory
        });
        containerMount
          .find("Box")
          .at(1)
          .simulate("click", mockEvent);
        expect(mockHistory.calledWithParam).toBe(Routes.url.browseListings);
      });
    });

    describe("handleLearn", () => {
      const mockEvent = {
        preventDefault: function() {
          this.preventDefaultCalled = true;
        },
        preventDefaultCalled: false
      };
      it("should call preventDefault on the event", () => {
        containerMount.setProps({ b: 1 });
        containerMount
          .find("Box")
          .at(2)
          .simulate("click", mockEvent);
        expect(mockEvent.preventDefaultCalled).toBe(true);
      });

      it("correct url is called", () => {
        const open = jest.spyOn(window, "open");
        containerMount
          .find("Box")
          .at(2)
          .simulate("click", mockEvent);
        expect(open).toHaveBeenCalledWith(
          Routes.externalLinks.learnRealEstate,
          "_blank"
        );
      });
    });
  });
});
