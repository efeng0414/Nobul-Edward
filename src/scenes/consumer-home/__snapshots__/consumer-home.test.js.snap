// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerHome Container Snapshot should match stored snapshot 1`] = `
<div>
  <div
    className="background home-page background--tint"
    style={
      Object {
        "backgroundColor": "",
        "backgroundImage": "url([object Object])",
      }
    }
  >
    <div
      className="background-content"
    >
      <h1
        className="homepage-title"
      >
        How can nobul help you?
      </h1>
      <div
        className="home-actions"
      >
        <div
          className="box box-home"
          onClick={[Function]}
        >
          <div
            className="box-icon blue"
          >
            <div
              className="box-icon-home"
            >
              <img
                className="primary-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="hover-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="check-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
            </div>
          </div>
          <h2>
            Buy or Sell
          </h2>
          <p>
            Find an agent who will help
          </p>
        </div>
        <div
          className="box box-home"
          onClick={[Function]}
        >
          <div
            className="box-icon purple"
          >
            <div
              className="box-icon-home"
            >
              <img
                className="primary-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="hover-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="check-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
            </div>
          </div>
          <h2>
            Browse Properties
          </h2>
          <p>
            See what is out there
          </p>
        </div>
        <div
          className="box box-home"
          onClick={[Function]}
        >
          <div
            className="box-icon yellow"
          >
            <div
              className="box-icon-home"
            >
              <img
                className="primary-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="hover-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
              <img
                className="check-icon"
                src={
                  Object {
                    "default": "file-mock",
                  }
                }
              />
            </div>
          </div>
          <h2>
            Learning Center
          </h2>
          <p>
            Got questions? We have answers
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="proof-banner"
  >
    <div
      className="proof-banner__inner"
    >
      <h5
        className="proof-banner__text"
      >
        <p
          className="proof-banner__text-item"
        >
          Hundreds of licensed agents looking to help
        </p>
        <p
          className="proof-banner__text-item"
        >
          Thousands of active buyers and sellers
        </p>
        <p
          className="proof-banner__text-item"
        >
          200,000+ listings across North America
        </p>
      </h5>
      <div
        className="proof-banner__logos"
      >
        <img
          alt="narcity"
          className="proof-banner__logo proof-banner__logo--ctv"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
        <img
          alt="Toronto Star"
          className="proof-banner__logo proof-banner__logo--toronto-star"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
        <img
          alt="Bloomberg"
          className="proof-banner__logo proof-banner__logo--bloomberg"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
        <img
          alt="Financial Post"
          className="proof-banner__logo proof-banner__logo--financial-post"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
        <img
          alt="tech crunch"
          className="proof-banner__logo proof-banner__logo--globe-and-mail"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
        <img
          alt="citybizlist"
          className="proof-banner__logo proof-banner__logo--citybizlist"
          src={
            Object {
              "default": "file-mock",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
